File: 프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.java
class Solution {
    public String solution(String my_string, String overwrite_string, int s) {
        String answer = "";
        
        StringBuilder sb = new StringBuilder(my_string);
        
        for(int i = 0; i < my_string.length(); i++) {
            if(i >= s && i < s + overwrite_string.length())
                sb.setCharAt(i, overwrite_string.charAt(i-s));
        }
        
        answer = sb.toString();
        
        return answer;
    }
}

File: 프로그래머스/0/181943. 문자열 겹쳐쓰기/문자열 겹쳐쓰기.cs
using System;
using System.Text;

public class Solution {
    public string solution(string my_string, string overwrite_string, int s) {
        string answer = "";
        
        StringBuilder sb = new StringBuilder();
        
        sb.Append(my_string.Substring(0, s)); // front
        
        for(int i = 0; i < overwrite_string.Length; i++) {
            sb.Append(overwrite_string[i]);
        }
        
        sb.Append(my_string.Substring(s+overwrite_string.Length)); // back
        
        answer = sb.ToString();
        
        return answer;
    }
}